plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergV2.java

```diff
@@ -256,6 +258,62 @@ public void testUnsettingAllTableProperties()
         assertQuery("SELECT * FROM " + tableName, "SELECT * FROM nation");
     }
 
+    @Test
+    public void testDeletingEntireFile()
+    {
+        String tableName = "test_deleting_entire_file_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT * FROM tpch.tiny.nation WITH NO DATA", 0);
+        assertUpdate("INSERT INTO " + tableName + " SELECT * FROM tpch.tiny.nation WHERE regionkey = 1", "SELECT count(*) FROM nation WHERE regionkey = 1");
+        assertUpdate("INSERT INTO " + tableName + " SELECT * FROM tpch.tiny.nation WHERE regionkey != 1", "SELECT count(*) FROM nation WHERE regionkey != 1");
+
+        assertThat(this.loadTable(tableName).newScan().planFiles()).hasSize(2);
+        assertUpdate("DELETE FROM " + tableName + " WHERE regionkey <= 2", "SELECT count(*) FROM nation WHERE regionkey <= 2");
+        assertQuery("SELECT * FROM " + tableName, "SELECT * FROM nation WHERE regionkey > 2");
+        assertThat(this.loadTable(tableName).newScan().planFiles()).hasSize(1);
+    }
+
+    @Test
+    public void testDeletingEntireFileWithNonTupleDomainConstraint()
+    {
+        String tableName = "test_deleting_entire_file_with_non_tuple_domain_constraint" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT * FROM tpch.tiny.nation WITH NO DATA", 0);
+        assertUpdate("INSERT INTO " + tableName + " SELECT * FROM tpch.tiny.nation WHERE regionkey = 1", "SELECT count(*) FROM nation WHERE regionkey = 1");
+        assertUpdate("INSERT INTO " + tableName + " SELECT * FROM tpch.tiny.nation WHERE regionkey != 1", "SELECT count(*) FROM nation WHERE regionkey != 1");
+
+        assertThat(this.loadTable(tableName).newScan().planFiles()).hasSize(2);
+        assertUpdate("DELETE FROM " + tableName + " WHERE regionkey % 2 = 1", "SELECT count(*) FROM nation WHERE regionkey % 2 = 1");
+        assertQuery("SELECT * FROM " + tableName, "SELECT * FROM nation WHERE regionkey % 2 = 0");
+        assertThat(this.loadTable(tableName).newScan().planFiles()).hasSize(1);
+    }
+
+    @Test
+    public void testDeletingEntireFileWithMultipleSplits()
+    {
+        String tableName = "test_deleting_entire_file_with_multiple_splits" + randomTableSuffix();
+        assertUpdate(
+                Session.builder(getSession()).setCatalogSessionProperty("iceberg", "orc_writer_max_stripe_rows", "5").build(),
+                "CREATE TABLE " + tableName + " WITH (format = 'ORC') AS SELECT * FROM tpch.tiny.nation", 25);
+        this.loadTable(tableName).updateProperties().set(SPLIT_SIZE, "100").commit();
```

Comment:
what for?

is 100 ok numher? we have 25 rows only

Reply:
```
private long getQuerySplits(QueryId queryId)
    {
        QueryStats stats = getDistributedQueryRunner().getCoordinator().getQueryManager().getFullQueryInfo(queryId).getQueryStats();
        long numberOfSplits = stats.getOperatorSummaries()
                .stream()
                .filter(summary -> summary.getOperatorType().equals("ScanFilterAndProjectOperator"))
                .mapToLong(OperatorStats::getTotalDrivers)
                .sum();
        return numberOfSplits;
    }
```

```
 ResultWithQueryId<MaterializedResult> deletionResult = getDistributedQueryRunner().executeWithQueryId(getSession(), "DELETE FROM " + tableName + " WHERE regionkey < 10");
        long deletionSplits = getQuerySplits(deletionResult.getQueryId());
```

I was hoping to see in the query stats that there are multiple splits for the file, but this wasn't the case.
I checked via debug and indeed there are actually ~ 62 splits of maximum `100B`.

Any idea how we could retrieve the number of splits in the test case?

Reply:
There are some Delta tests that check the number of splits, but they are a bit finicky / often flaky so I didn't include one here.